### DISCUSSION TITLE
Change probability threshold in prediction

### DISCUSSION THREAD

**Discussion by agatadefr** [MIC-DKFZ/nnUNet#922] [General]
Hello,

I would like to change  threshold in prediction to detect more lesions ? 
Where could I do that in the code ? 

I would appreciate for any suggestion.

**Comment by FabianIsensee** [comment]
Hi, you would need to change the way the segmentation is obtained from the softmax probabilities. 

Take a look at https://github.com/MIC-DKFZ/nnUNet/blob/f5e72b1e4289c69af86108107e98ac3332ae8323/nnunet/inference/predict.py#L131

In here we use multi processing to export the segmentations. Exporting first does resampling, then argmax and finally the saving of the result:
https://github.com/MIC-DKFZ/nnUNet/blob/f5e72b1e4289c69af86108107e98ac3332ae8323/nnunet/inference/predict.py#L262

So if you want to change the way this is done you need to change `predict_cases` and also `save_segmentation_nifti_from_softmax` (the latter is here: https://github.com/MIC-DKFZ/nnUNet/blob/f5e72b1e4289c69af86108107e98ac3332ae8323/nnunet/inference/segmentation_export.py#L27)
